### PR TITLE
ref: Release archives for everybody

### PR DESCRIPTION
**Pull request by untitaker**
The option is True in prod for a few days now.

Wasn't sure if we want to keep the min-count but I assume yes.

Fix [INGEST-39]

[INGEST-39]: https://getsentry.atlassian.net/browse/INGEST-39

### CODE REVIEW

File: src/sentry/tasks/assemble.py [hunk 1]
```diff
@@ -295,22 +295,11 @@ def assemble_artifacts(org_id, version, checksum, chunks, **kwargs):
...
-
-            # NOTE(jjbayer): Single files are still stored to enable
-            # rolling back from release archives. Once release archives run
-            # smoothely, this call can be removed / only called when feature
-            # flag is off.
-            count_as_artifacts = not saved_as_archive
-            _store_single_files(archive, meta, count_as_artifacts)
+            min_size = options.get("processing.release-archive-min-files")
+            if num_files >= min_size:
+                update_artifact_index(release, dist, bundle)
```
Comment (jjbayer) [inline src/sentry/tasks/assemble.py:300]: Not 100% sure if we can remove the catch-all here, but there haven't been any issues in the past few days apparently [1], so I guess it's fine.

[1] https://sentry.io/organizations/sentry/issues/?groupStatsPeriod=auto&query=%22Unable+to+update+artifact+index%22&statsPeriod=14d

**Comment by jjbayer** [comment]
We should also remove the [option check](https://github.com/getsentry/sentry/blob/master/src/sentry/lang/javascript/processor.py#L575-L580) in processing, because we cannot rely on individual files here anymore:

```python
        sample_rate = options.get("processing.use-release-archives-sample-rate")
        if sample_rate and random.random() < sample_rate:
            # Read from archive
            result = fetch_release_artifact(url, release, dist)
        else:
            result = fetch_release_file(url, release, dist)
```

File: src/sentry/tasks/assemble.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/tasks/assemble.py]: You have to, right? because you don't want to store single files anymore so you don't have a fallback path anyway

Comment (untitaker) [reply src/sentry/tasks/assemble.py]: ah I see, okay

**Review by untitaker — COMMENTED** [review]
changes by joris lgtm

**Review by jjbayer — APPROVED** [review]

**Pull request merged**